### PR TITLE
Profile Screen

### PR DESCRIPTION
**Pull request by imashnake0**
[comment]: # (Replace [ ] with [x].)
- [x] Read [`CONTRIBUTING.md`](https://github.com/imashnake0/Animite/blob/be53ba4561c8ff20f588fb348965f208e301b6b8/CONTRIBUTING.md).

**Summary of changes:**

1. Added a basic banner layout.
2. Use [Compose Markdown](https://github.com/boswelja/compose-markdown) to add user's `about`.
3. Added tabs.

**Tested changes:**

I

[comment]: # (THANK YOU! ♡\(灬♥ ◡ ♥灬\))

### CODE REVIEW

File: api/anilist/src/main/kotlin/com/imashnake/animite/api/anilist/AnilistMediaRepository.kt [hunk 1]
```diff
@@ -42,7 +42,7 @@ class AnilistMediaRepository @Inject constructor(
                     seasonYear = Optional.presentIfNotNull(seasonYear)
                 )
             )
-            .fetchPolicy(FetchPolicy.CacheAndNetwork)
+            .fetchPolicy(FetchPolicy.CacheFirst)
```
Comment (boswelja) [inline api/anilist/src/main/kotlin/com/imashnake/animite/api/anilist/AnilistMediaRepository.kt:45]: Separate PR :pray:

File: app/src/main/java/com/imashnake/animite/features/home/HomeScreen.kt [hunk 2]
```diff
@@ -61,6 +61,7 @@ import com.ramcosta.composedestinations.navigation.DestinationsNavigator
 import com.imashnake.animite.R
 import com.imashnake.animite.core.ui.LocalPaddings
 import com.imashnake.animite.core.ui.layouts.BannerLayout
+import com.imashnake.animite.core.R as coreR
```
Comment (boswelja) [inline app/src/main/java/com/imashnake/animite/features/home/HomeScreen.kt:64]: Oh no not again

File: core/src/main/kotlin/com/imashnake/animite/core/extensions/MarkdownStyleExt.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+@Composable
+fun animiteBlockQuoteStyle(
+    background: Color = MaterialTheme.colorScheme.surfaceVariant,
+    barWidth: Dp = dimensionResource(R.dimen.markdown_block_corner_radius),
+    barColor: Color = MaterialTheme.colorScheme.onSurfaceVariant,
+    barShape: Shape = RectangleShape,
+    innerPadding: PaddingValues = PaddingValues(8.dp)
+) = BlockQuoteStyle(
+    background = background,
+    shape = RoundedCornerShape(barWidth),
```
Comment (boswelja) [inline core/src/main/kotlin/com/imashnake/animite/core/extensions/MarkdownStyleExt.kt:26]: Not gonna define this as part of the theme?

File: core/src/main/kotlin/com/imashnake/animite/core/extensions/MarkdownStyleExt.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    barShape = barShape,
+    barWidth = barWidth,
+    barColor = barColor,
+    innerPadding = innerPadding
+)
+
+@Composable
+fun animiteCodeBlockStyle(
+    background: Color = MaterialTheme.colorScheme.surfaceVariant,
+    shape: Shape = RoundedCornerShape(dimensionResource(R.dimen.markdown_block_corner_radius)),
```
Comment (boswelja) [inline core/src/main/kotlin/com/imashnake/animite/core/extensions/MarkdownStyleExt.kt:36]: Why are we passing a Dp in one and a Shape in the other

File: gradle/libs.versions.toml [hunk 5]
```diff
@@ -79,6 +79,10 @@ junit = "4.13.2"
 # https://github.com/detekt/detekt/releases.
 detekt = "1.23.4"
 
+# Compose Markdown
+# https://github.com/boswelja/compose-markdown.
+composeMarkdown = "1.0.3"
```
Comment (boswelja) [inline gradle/libs.versions.toml:84]: Brace for 1.0.4

File: profile/src/main/kotlin/com/imashnake/animite/profile/ProfileScreen.kt [hunk 6]
```diff
@@ -2,24 +2,52 @@ package com.imashnake.animite.profile
...
+import com.imashnake.animite.core.ui.LocalPaddings
+import com.imashnake.animite.core.ui.NestedScrollableContent
+import com.imashnake.animite.core.ui.layouts.BannerLayout
 import com.imashnake.animite.profile.dev.internal.ANILIST_AUTH_DEEPLINK
 import com.ramcosta.composedestinations.annotation.DeepLink
 import com.ramcosta.composedestinations.annotation.Destination
 import com.ramcosta.composedestinations.annotation.RootNavGraph
+import com.imashnake.animite.core.R as coreR
 
+@Suppress("LongMethod")
```
Comment (boswelja) [inline profile/src/main/kotlin/com/imashnake/animite/profile/ProfileScreen.kt:50]: Make it not long I dare you

File: profile/src/main/kotlin/com/imashnake/animite/profile/ProfileScreen.kt [hunk 7]
```diff
@@ -44,23 +72,131 @@ fun ProfileScreen(
...
+                                overflow = TextOverflow.Ellipsis,
+                                modifier = Modifier.padding(horizontal = LocalPaddings.current.large)
+                            )
+                            Box(
+                                Modifier
+                                    .maxHeight(dimensionResource(R.dimen.user_about_height))
+                                    .padding(horizontal = LocalPaddings.current.large)
+                            ) {
+                                NestedScrollableContent { contentModifier ->
+                                    about?.let {
```
Comment (boswelja) [inline profile/src/main/kotlin/com/imashnake/animite/profile/ProfileScreen.kt:114]: Put the let outside of the Box and NestedScrollableContent

File: profile/src/main/kotlin/com/imashnake/animite/profile/ProfileScreen.kt [hunk 8]
```diff
@@ -44,23 +72,131 @@ fun ProfileScreen(
...
+            titles.forEachIndexed { index, title ->
+                Tab(
+                    selected = state == index,
+                    onClick = { state = index },
+                    text = {
+                        Text(
+                            text = title,
+                            overflow = TextOverflow.Ellipsis,
+                            style = MaterialTheme.typography.bodyMedium,
+                            color = MaterialTheme.colorScheme.onBackground.copy(
```
Comment (boswelja) [inline profile/src/main/kotlin/com/imashnake/animite/profile/ProfileScreen.kt:168]: Hoist onBackground color so there's a single value to change to update the background for this and the gradient

**Review by boswelja — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/com/imashnake/animite/core/extensions/MarkdownStyleExt.kt (hunk 3 above)
Comment (imashnake0) [reply core/src/main/kotlin/com/imashnake/animite/core/extensions/MarkdownStyleExt.kt:26]: [Shape.kt](https://github.com/imashnake0/Animite/blob/main/app/src/main/java/com/imashnake/animite/features/theme/Shape.kt) is currently empty, I'm in the process of creating components in Figma so we'll have a more solid design system soon. I'll fix typography, shapes, colors, and themes in a follow up PR.

File: core/src/main/kotlin/com/imashnake/animite/core/extensions/MarkdownStyleExt.kt (hunk 4 above)
Comment (imashnake0) [reply core/src/main/kotlin/com/imashnake/animite/core/extensions/MarkdownStyleExt.kt:36]: Markdown code blocks in Animite will have a shape that is determined by the width of the bar: 

Screen Shot 2024-02-27 at 4 28 47 AM

(Yes, I take you guys very seriously 👽)

File: profile/src/main/kotlin/com/imashnake/animite/profile/ProfileScreen.kt (hunk 6 above)
Comment (imashnake0) [reply profile/src/main/kotlin/com/imashnake/animite/profile/ProfileScreen.kt:50]: I tried (61/60) 😔, things will get shorter once I introduce design system components tho

**Review by boswelja — APPROVED** [review]

**Pull request merged**